src/plugins/janus_sip.c

```diff
@@ -1771,6 +1772,127 @@ static void janus_sip_sofia_logger(void *stream, char const *fmt, va_list ap) {
 	}
 }
 
+/* Sofia SIP logger function: when the Event Handlers mechanism is enabled,
+ * we use this to intercept SIP messages sent by the stack (received
+ * messages are more easily recoverable in janus_sip_sofia_callback) */
+static void janus_sip_sofia_logger_single(void *stream, char const *fmt, va_list ap) {
+	if(!fmt)
+		return;
+	char line[255];
+	char message[16000];
+	char sofia_log_buf[16000];
+	char call_id_hdr[255];
+	sofia_log_buf[0] = '\0';
+	call_id_hdr[0] = '\0';
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#ifndef __clang__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-attribute=format"
+#endif
+	g_vsnprintf(line, sizeof(message), fmt, ap);
+#ifndef __clang__
+#pragma GCC diagnostic pop
+#endif
+#pragma GCC diagnostic pop
+	gchar **parts = g_strsplit(message, "\n", -1);
+	if (parts) {
+		int index = 0;
+		char *line = NULL, *cr = NULL;
+		while((line = parts[index]) != NULL) {
+			cr = strchr(line, '\r');
+			if(cr != NULL)
+				*cr = '\0';
+			if(*line == '\0') {
+				if(cr != NULL)
+					*cr = '\r';
+				index++;
+				continue;
+			}
+
+			/* First line of the log indicates if we send or receive a message. */
+			/* We are only interested in the sent ones */
+			if(index==0) {
+				if(strstr(line, "send ") == line) {
+					int length = atoi(line+5);
+					JANUS_LOG(LOG_HUGE, "Intercepting message (%d bytes)\n", length);
+					index++;
+					continue;
+				}
+				g_strfreev(parts);
+				return ;
+			}
+
+			if(strlen(line) == 1) {
+				/* Append a carriage and return */
+				janus_strlcat(sofia_log_buf, "\r\n", sizeof(sofia_log_buf));
+			} else if(strstr(line, "-") == line ) {
+				index++;
+				continue;
+			} else if(strstr(line, "OPTIONS") == line) {
+				index++;
+				continue;
+			} else {
+				/* Is this a Call-ID header? Keep note of it */
+				if(strstr(line, "Call-ID") == line) {
+					g_snprintf(call_id_hdr, sizeof(call_id_hdr), "%s", line+9);
```

Comment:
Why is this 9? Are you assuming `Call-ID: `? The `strstr` call above is looking for `Call-ID`, not `Call-ID: `, which can cause problems if the header is formatted differently, or if `Call-ID` is not followed by a semicolon.

Reply:
This also existed in janus_sip_sofia_logger. I suppose it's safe to assume that "Call-ID: " will always be sent with this format,  as it is Sofia-SIP library the one that generates those messages and we only process the messages sent from Sofia SIP stack.

Reply:
Makes sense!